sdk/spring/spring-cloud-azure-feature-management/README.md

```diff
@@ -192,10 +192,159 @@ feature-management:
+                  type: "Numbered"
+                  numberOfOccurrences: 3
+```
+
+The `recurrence` settings is made up of two parts: `pattern` (how often the time window will repeat) and `range` (for how long the recurrence pattern will repeat). To create a recurrence rule, you must specify both `pattern` and `range`. Any pattern type can work with any range type.
+
+**Advanced:** The time zone offset of the `start` property will apply to the recurrence settings.
+
+#### Recurrence Pattern
+
+There are two possible recurrence pattern types: `Daily` and `Weekly`. For example, a time window could repeat "every day", "every 3 days", "every Monday" or "every other Friday".
+
+* Parameters
+
+  Property | Relevance | Description
+  -----------|-------|-----
+  **type** | Required  | Two valid values: `Daily`, `Weekly`.
+  **interval** | Optional  | Specifies the interval between each start time of occurrence. Default value is 1. <br/>For example, if the interval is 2 with `Daily` type, and current occurrence is 2:00 AM ~ 3:00 AM on 2024/05/11, then the next occurrence should be 2:00 AM ~ 3:00 AM on 2024/05/13
+  **daysOfWeek** | Optional  | Specifies on which day(s) of the week the event occurs. It's required when `Weekly` type, not applicable for `Daily` type. 
+  **firstDayOfWeek** | Optional  | Specifies which day is considered the first day of the week. Default value is `Sunday`. Not applicable for `Daily` type. 
+  
+  * Example
+    * Daily
+  
+        The daily recurrence pattern causes the time window to repeat based on interval between each occurrence.
+
+        The following example will repeat from 2:00 AM to 3:00 AM on every 2 days
+
+        ```yaml
+        start: "Mon, 13 May 2024 02:00:00 GMT",
+        end: "Mon, 13 May 2024 03:00:00 GMT",
+        recurrence:
+          pattern:
+            type: "Daily",
+            interval: 2
+          range:
+            type: "NoEnd"
+        ```
+
+    * Weekly
+  
+      The weekly recurrence pattern causes the time window to repeat on the same day or days of the week, based on the number of weeks between each set of occurrences.
+
+      The following example will repeat from 2:00 AM to 3:00 AM on every other Monday and Tuesday
+
+        ```yaml
+        start: "Mon, 13 May 2024 02:00:00 GMT",
+        end: "Mon, 13 May 2024 03:00:00 GMT",
+        recurrence:
+          pattern:
+            type: "Weekly",
+            interval: 2,
+            daysOfWeek: 
+              - Monday
+              - Tuesday
+          range:
+            type: "NoEnd"
+        ```
+
+    **Note:** `start` must be a valid first occurrence which fits the recurrence pattern. For example, if we define to repeat on every other Monday and Tuesday, then the start time should be in Monday or Tuesday. <br /> Additionally, the duration of the time window cannot be longer than how frequently it occurs. For example, it is invalid to have a 25-hour time window recur every day.
+
+#### Recurrence Range
```

Comment:
Is range itself a required field? I can currently make a time window filter without one.

Reply:
Yes, it's a required field. And there's a validation in `validateRecurrenceRequiredParameter` of "RecurrenceValidator.java" file, if range is null, will throw an exception.
https://github.com/ivywei0125/azure-sdk-for-java/blob/yuwe/recurring-time-window-filter/sdk/spring/spring-cloud-azure-feature-management/src/main/java/com/azure/spring/cloud/feature/management/implementation/timewindow/recurrence/RecurrenceValidator.java

Reply:
Have you tested this? I have no exception is thrown for me when I make one with it.

Reply:
I had a test case(`generalRequiredParameterTest`) for testing this case, but the test case didn't run as expected. 
https://github.com/ivywei0125/azure-sdk-for-java/blob/yuwe/recurring-time-window-filter/sdk/spring/spring-cloud-azure-feature-management/src/test/java/com/azure/spring/cloud/feature/management/filters/recurrence/RecurrenceValidatorTest.java

Before, I use the following code to catch the exception and then check the exception type and error message. But if there's no exception, the test case also succeeded while this is not as expected.
```
try {
    final TimeWindowFilter filter = new TimeWindowFilter();
    final FeatureFilterEvaluationContext context = new FeatureFilterEvaluationContext();
    context.setParameters(parameters);
    filter.evaluate(context);
} catch (final Exception e) {
     assertTrue(e instanceof IllegalArgumentException);
     assertTrue(e.getMessage().contains(errorMessage));
}
```

Updated to use `assertThrows`, so the test case will fail when no exception.
```
final TimeWindowFilter filter = new TimeWindowFilter();
final FeatureFilterEvaluationContext context = new FeatureFilterEvaluationContext();

final IllegalArgumentException exception = assertThrows(
    IllegalArgumentException.class,
    () -> {
        context.setParameters(parameters);
        filter.evaluate(context);
    });
assertTrue(exception.getMessage().contains(errorMessage));
```

Thanks for testing for this. I haven't run it in a real application. I just try to verify this by test case. But it turned out that the test case is wrong.